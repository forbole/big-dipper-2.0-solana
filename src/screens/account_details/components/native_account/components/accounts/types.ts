export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
