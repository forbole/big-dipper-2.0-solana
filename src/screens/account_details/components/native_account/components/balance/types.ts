export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
