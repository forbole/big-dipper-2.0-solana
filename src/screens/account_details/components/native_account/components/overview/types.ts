export type OverviewType = {
  address: string;
}
