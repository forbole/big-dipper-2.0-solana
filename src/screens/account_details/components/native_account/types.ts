export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
