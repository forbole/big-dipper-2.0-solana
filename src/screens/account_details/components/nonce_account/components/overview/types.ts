export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
