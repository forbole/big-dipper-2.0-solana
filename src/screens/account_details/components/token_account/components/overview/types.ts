export type OverviewType = {
  address: string;
  mint: string;
  balance: string;
  authority: string;
}
