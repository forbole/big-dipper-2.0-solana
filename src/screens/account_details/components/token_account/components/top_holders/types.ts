export type DataType = {
  address: string;
  amount: TokenUnit;
  value: number;
}

export type HoldersState = {
  loading: boolean;
  exists: boolean;
  data: DataType[];
}
