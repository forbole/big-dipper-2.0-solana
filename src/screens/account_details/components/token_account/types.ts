export type HeaderType = {
  token: string;
  mint: string;
  imageUrl: string;
}
export type OverviewType = {
  mint: string;
  decimals: number;
  mintAuthority: string;
  freezeAuthority: string;
}

export type MarketType = {
  price: number;
  marketCap: number;
  supply: number;
  holders: number;
  transactions: number;
}

export type HolderType = {
  address: string;
  quantity: number;
  percentage: number;
  value: number;
}

export type TokenDetailState = {
  loading: boolean;
  exists: boolean;
  header: HeaderType;
  overview: OverviewType;
  market: MarketType;
  holders: {
    hasNextPage: boolean;
    isNextPageLoading: boolean;
    offsetCount: number;
    data: HolderType[];
  };
  transactions: {
    hasNextPage: boolean;
    isNextPageLoading: boolean;
    offsetCount: number;
    data: Transactions[];
  };
}
