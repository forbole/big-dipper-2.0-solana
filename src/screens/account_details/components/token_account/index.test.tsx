/* eslint-disable */
import React from 'react';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import { TokenDetailsDocument } from '@graphql/types';
import BlockDetails from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      height: 300,
    },
  }),
}));

jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
  LoadAndExist: (props) => <div id="LoadAndExist" {...props} />,
}));

jest.mock('./components', () => ({
  Header: (props) => <div id="Header" {...props} />,
  Overview: (props) => <div id="Overview" {...props} />,
  // Transactions: (props) => <div id="Transactions" {...props} />,
  TopHolders: (props) => <div id="TopHolders" {...props} />,
  Market: (props) => <div id="Market" {...props} />,
}));

jest.mock('..', () => ({
  Transactions: (props) => <div id="Transactions" {...props} />,
}));

const mockData = jest.fn().mockResolvedValue({
  "data": {
    "tokenUnit": [
      {
        "mint": "SBTCB6pWqeDo6zGi9WVRMLCsKsN6JiR1RMUqvLtgSRv",
        "unitName": "Saber Wrapped Bitcoin (Sollet) (8 decimals)",
        "logo": "https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/SBTCB6pWqeDo6zGi9WVRMLCsKsN6JiR1RMUqvLtgSRv/icon.png",
        "tokenPrice": {
          "marketCap": 807393897904,
          "price": 42485
        }
      }
    ],
    "token": [
      {
        "mint": "SBTCB6pWqeDo6zGi9WVRMLCsKsN6JiR1RMUqvLtgSRv",
        "freezeAuthority": "",
        "decimals": 8,
        "mintAuthority": "GpkFF2nPfjUcsavgDGscxaUEQ2hYJ563AXXtU8ohiZ7c"
      }
    ],
    "tokenSupply": [
      {
        "supply": 28421156478
      }
    ],
    "tokenAccountAggregate": {
      "aggregate": {
        "count": 3355
      }
    }
  }
});

// ==================================
// unit tests
// ==================================
describe('screen: BlockDetails', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      TokenDetailsDocument,
      mockData,
    );

    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <BlockDetails />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
