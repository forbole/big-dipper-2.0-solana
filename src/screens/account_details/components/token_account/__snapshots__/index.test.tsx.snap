// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: BlockDetails matches snapshot 1`] = `
<div
  exists={true}
  id="LoadAndExist"
  loading={false}
>
  <div
    className="makeStyles-root"
  >
    <div
      className="makeStyles-header"
      header={
        Object {
          "imageUrl": "https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/SBTCB6pWqeDo6zGi9WVRMLCsKsN6JiR1RMUqvLtgSRv/icon.png",
          "mint": "SBTCB6pWqeDo6zGi9WVRMLCsKsN6JiR1RMUqvLtgSRv",
          "token": "Saber Wrapped Bitcoin (Sollet) (8 decimals)",
        }
      }
      id="Header"
    />
    <div
      className="makeStyles-overview"
      id="Overview"
      overview={
        Object {
          "decimals": 8,
          "freezeAuthority": "",
          "mint": "SBTCB6pWqeDo6zGi9WVRMLCsKsN6JiR1RMUqvLtgSRv",
          "mintAuthority": "GpkFF2nPfjUcsavgDGscxaUEQ2hYJ563AXXtU8ohiZ7c",
        }
      }
    />
    <div
      className="makeStyles-market"
      id="Market"
      market={
        Object {
          "holders": 3355,
          "marketCap": 807393897904,
          "price": 42485,
          "supply": 28421156478,
        }
      }
    />
    <div
      className="makeStyles-holders"
      id="TopHolders"
    />
  </div>
</div>
`;
